src/s3path.jl

```diff
@@ -193,6 +193,37 @@ function Base.isdir(fp::S3Path)
     return iterate(objects) !== nothing
 end
 
+function FilePathsBase.walkpath(fp::S3Path; topdown=true, follow_symlinks=false, onerror=throw)
+
+    # Select objects with that prefix
+    objects = s3_list_objects(fp.config, fp.bucket, fp.key; delimiter="")
+
+    # Extract keys from the objects and remove the prefix
+    rm_key = s -> chop(s, head=length(fp.key), tail=0)
+    keys = map(rm_key, o["Key"] for o in objects)
+
+    isempty(keys) && return Channel{S3Path}()
+
+    chnl = S3Path[]
```

Comment:
I tried this originally via a task but 1) I couldn't check if `_p in chnl` while the `Channel` was still open and 2) I don't think I can call `reverse` on the `Channel` either so this seemed the easiest way to proceed even though it uses more memory.